TravelAgency-persistence/src/main/java/cz/muni/fi/travelAgency/entities/Excursion.java

```diff
@@ -1,26 +1,186 @@
 package cz.muni.fi.travelAgency.entities;
 
+import java.math.BigDecimal;
+import java.util.Date;
+import java.util.Objects;
 import javax.persistence.*;
 import javax.validation.constraints.NotNull;
-import java.util.Objects;
+import javax.validation.constraints.DecimalMin;
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.Set;
+
+
 
+/**
+ *
+ * Representing Excursion Entity
+ *
+ * @author xrajivv
+ */
 @Entity
+@Table(name = "EXCURSION")
 public class Excursion {
 
+    /**
+     * Unique ID set by the DB
+     */
     @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
+    /**
+     * Description of the Excursion
+     */
+    @NotNull
+    @Column(nullable = false)
+    private String description;
+    /**
+     * Destination of the Excursion
+     */
+    @NotNull
+    @Column(nullable = false)
+    private String destination;
+    /**
+     * Price of the Excursion
+     */
+    @DecimalMin("0.0")
+    @NotNull
+    @Column(nullable = false)
+    private BigDecimal price;
+    /**
+     * Date of the Excursion
+     */
+    @NotNull
+    @Column(nullable = false)
+    @Temporal(TemporalType.DATE)
+    private Date excursionDate;
+    /**
+     * Duration of the Excursion
+     */
+    @NotNull
+    @Column(nullable = false)
+    private Duration excursionDuration;
+    
+    @ManyToOne
```

Comment:
I think should be OneToMany

Reply:
working on it

Reply:
I think it should be trip instead of Set of trips and annotation should be ManyToOne